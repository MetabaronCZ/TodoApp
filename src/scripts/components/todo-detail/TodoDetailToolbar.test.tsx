import React from 'react';

import { render } from '@testing-library/react';
import { describe, expect, it } from '@jest/globals';

import { TodoDetailToolbar } from 'components/todo-detail/TodoDetailToolbar';
import { withMockedProviders } from 'test/component';

describe('components/todo-detail/TodoDetailToolbar', () => {
  it('should render correctly', () => {
    const { container } = render(withMockedProviders(<TodoDetailToolbar />));
    expect(container).toMatchSnapshot();
  });

  it('should be disabled', () => {
    const { container } = render(
      withMockedProviders(<TodoDetailToolbar disabled />),
    );
    expect(container).toMatchSnapshot();
    expect(container.querySelector('button:disabled')).toBeInTheDocument();
  });
});
