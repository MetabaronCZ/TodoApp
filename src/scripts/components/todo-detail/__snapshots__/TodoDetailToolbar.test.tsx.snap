// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/todo-detail/TodoDetailToolbar > should be disabled 1`] = `
.c3 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: block;
  width: 20px;
  height: 20px;
  line-height: 20px;
  font-style: normal;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  display: block;
  width: 100%;
  border: none;
  outline: none;
  cursor: pointer;
}

.c2 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
  padding: 5px 10px;
  padding-left: 5px;
  background: #f0f0f0;
}

.c2:hover:not(:disabled) {
  background-color: #e0e0e0;
}

.c2:focus {
  outline: 1px dotted #222;
  outline-offset: -1px;
}

.c2:disabled {
  color: #c0c0c0;
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 1000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
  height: calc(30px + 1px);
  border-bottom: 1px solid #c0c0c0;
  background: #f0f0f0;
}

<div>
  <div
    class="c0"
  >
    <div>
      <button
        class="c1 c2"
        disabled=""
        type="button"
      >
        <i
          class="c3"
          style="color: rgb(192, 192, 192);"
        >
          ❮
        </i>
        <span
          class="c4"
        >
          Zpět
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`components/todo-detail/TodoDetailToolbar > should render correctly 1`] = `
.c2 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: block;
  width: 20px;
  height: 20px;
  line-height: 20px;
  font-style: normal;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
  padding: 5px 10px;
  padding-left: 5px;
  background: #f0f0f0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover:not(:disabled) {
  background-color: #e0e0e0;
}

.c1:focus {
  outline: 1px dotted #222;
  outline-offset: -1px;
}

.c1:disabled {
  color: #c0c0c0;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 1000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
  height: calc(30px + 1px);
  border-bottom: 1px solid #c0c0c0;
  background: #f0f0f0;
}

<div>
  <div
    class="c0"
  >
    <div>
      <a
        class="c1"
        href="/"
      >
        <i
          class="c2"
        >
          ❮
        </i>
        <span
          class="c3"
        >
          Zpět
        </span>
      </a>
    </div>
  </div>
</div>
`;
